packages/sbcl/build.sh

```diff
@@ -0,0 +1,112 @@
+TERMUX_PKG_HOMEPAGE="http://www.sbcl.org/"
+TERMUX_PKG_DESCRIPTION="A high performance Common Lisp compiler"
+TERMUX_PKG_LICENSE="custom"
+TERMUX_PKG_LICENSE_FILE="COPYING"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="2.5.3"
+TERMUX_PKG_SRCURL="https://github.com/sbcl/sbcl/archive/refs/tags/sbcl-${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256=f3b923778f7d1f151f2bdc08a0e92ec4f5a3db5efca5f46ea2ac439dda06cc35
+TERMUX_PKG_DEPENDS="zstd"
+# TERMUX_ON_DEVICE_BUILD=true  build dependencies: ecl, strace
+# TERMUX_ON_DEVICE_BUILD=false build dependencies: aosp-libs, bash, coreutils, strace
+# NOTE: if you are bootstrapping on-device, if either is desired, it is possible to manually
+# implement both an on-device build and an x86 build of ecl, but those are currently
+# unimplemented in termux-packages at time of writing.
+# the build dependency is disabled to avoid dependency errors when cross-compiling in CI for x86.
+TERMUX_PKG_BUILD_DEPENDS="aosp-libs, bash, coreutils, strace"
+# it is extremely difficult to port SBCL to 32-bit Android because of multiple severe
+# incompatibilties, like the lack of support for a fully position-independent runtime executable.
+TERMUX_PKG_EXCLUDED_ARCHES="arm, i686"
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_HOSTBUILD=true
+
+termux_step_host_build() {
+	if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
```

Comment:
So, parts of this build only apply for on-device builds and others only apply for cross-builds?
```suggestion
	if [[ "$TERMUX_ON_DEVICE_BUILD" == "true" ]]; then
```

Reply:
> So, parts of this build only apply for on-device builds and others only apply for cross-builds?

That is correct, and both are fully tested, however for now the on-device-only dependency `ecl` is commented out here because the `ecl` package in termux does not currently provide x86 builds, but I have written this to support cross-compiling `sbcl` for x86_64 (64-bit Android-x86) as well as aarch64.

It is possible to manually set up a messy build of `ecl` for Android-x86, then use that to test the on-device build of `sbcl` for Android-x86, so I have done that locally to know that build codepath is working.

I might make another PR in the future to improve the preexisting `ecl` package by trying to write and enable x86 targets for it, and after that is done, `ecl` could be explicitly added to the build dependencies of `sbcl` to allow seamless use of the on-device build codepath without a dependency error appearing in the cross-compilion codepath for x86, but I am just keeping this PR focused on `sbcl` because of the relatively large amount of changes necessary to initially enable this package.

> modern test condition syntax

sure, I will switch this file to that, wait a little while.